flow/project.py

```diff
@@ -912,9 +1016,11 @@ def isdisjoint(self, group):
         """
         return set(self).isdisjoint(set(group))
 
-    def _generate_id(self, job, operation_name=None, index=0):
+    def _generate_id(self, jobs, operation_name=None, index=0):
         "Return an id, which identifies this group with respect to this job."
-        project = job._project
+        project = jobs[0]._project
+
+        project._verify_aggregate_project(jobs)
```

Comment:
This check is unnecessary as it is internal.

Reply:
@b-butler 
Since aggregator function is user-facing.
I think this check is necessary as user can yield jobs from a different project inside that method.

Reply:
Is `_generate_id` the right place to check this though then? If we are worried about aggregation should we not check closer to the aggregation? Even if it requires the function in a couple more places, it will be more clear to the developer as to why the check exists in the first place.